### PR TITLE
🦊 Metamask integration POC

### PR DESCRIPTION
**Pull request by thesan**
[Demo](https://pioneer-2-git-feature-metamask-snap-joystream.vercel.app/)

- `@chainsafe/polkadot-snap` **fails on FF** for me so the demo only works on Chrome based browsers AFAIK.
- ~~The demo points at my publish version of the snap, so it only works with [Metamask Flask](https://chromewebstore.google.com/detail/metamask-flask-developmen/ljfoeinjpaedjfecbmggjgodbgkmjkjk).~~
- The snap works with HTTP RPC endpoints only (adding WS support is a headache). So the Apps implementing this should start storing the HTTP RPC link too. ~~In the meantime it works with a simple text replacement hack.~~ Done :heavy_check_mark:

### CODE REVIEW

**Comment by thesan** [comment]
@kdembler the official snap seems to work now but it still reports an error when connecting. Also I it might have been working before the last release, but I was focussed on this error before realizing that the `wsRpcUrl` should actually be the http rpc url :confused:.

I reported the issue (with the error message): https://github.com/ChainSafe/metamask-snap-polkadot/issues/231

Going forward:
- Some more QA is needed especially with the real (i.e not flask) Metamask and with extrinsics that are Joystream specific.
- The apps supporting this should store the http rpc endpoints in addition to the ws one.
- Also maybe this should be enabled on Chrome based browser only for now. I had forgotten about this and got confused on FF today :facepalm:

**Comment by thesan** [comment]
@kdembler I now get the `0.9.0` version of the snap with MetaMask:
image

However it still doesn't work on FF for me. I get to this screen but nothing happens:
image
Can you confirm it works for you on FF ?

File: packages/ui/src/app/config/network.ts [hunk 1]
```diff
@@ -57,6 +60,13 @@ const NODE_RPC_ENDPOINT: PredefinedEndpoint = {
   'local-mocks': 'ws://127.0.0.1:9944',
 }
 
+const NODE_HTTP_RPC_ENDPOINT: PredefinedEndpoint = {
+  mainnet: MAINNET_NODE_HTTP_RPC,
+  local: 'http://127.0.0.1:9933', // TODO: check
+  testnet: TESTNET_NODE_HTTP_RPC,
+  'local-mocks': 'http://127.0.0.1:9933', // TODO: check
```
Comment (kdembler) [inline packages/ui/src/app/config/network.ts:67]: Do we need those TODOs?

**Comment by kdembler** [comment]
Walletconnect seems to not work for me after this change. When I select the option, I don't get the connect popup

File: packages/ui/src/common/providers/network-endpoints/provider.tsx [hunk 2]
```diff
@@ -21,9 +22,24 @@ const EndpointsSchema = Yup.object().shape({
...
+): NetworkEndpoints => ({
+  ...newEndpoints,
+  nodeHttpRpcEndpoint:
+    newEndpoints.nodeHttpRpcEndpoint ??
+    newEndpoints.nodeRpcEndpoint
+      .replace('ws:', 'http:')
+      .replace('wss:', 'https:')
+      .replace('ws-rpc', 'http-rpc')
+      .replace(':9944', ''),
+})
```
[the remark this answers is not in the document]
Comment (thesan) [reply packages/ui/src/common/providers/network-endpoints/provider.tsx:38]: Actually no but we should make sure that the HTTP endpoints are defined in the environment.
Also the code should get the endpoint from the playground configs (I assumed it wasn't in there so I left the logic but it is in the configs e.g: [https://atlas-next.joystream.org/network/config.json](https://atlas-next.joystream.org/network/config.json)).

**Review by kdembler — APPROVED** [review]

**Pull request merged**